src/building/building_state.c

```diff
@@ -60,6 +60,15 @@ static void write_type_data(buffer *buf, const building *b)
         for (int i = 0; i < 8; i++) {
             buffer_write_u8(buf, 0);
         }
+        if(b->type == BUILDING_CARAVANSERAI) {
```

Comment:
Yes but building_has_supplier_inventory is true too for caravanserai, so I don't enter in the else building_monument_type_is_monument condition.

Reply:
Hummm. I see. This goes hand in hand in what I say below, a building should either be a supplier or a monument.

This is something for me to fix later on (it will require a new savegame version), for now leave this as-is.